### PR TITLE
Implement debugpy for python

### PR DESCRIPTION
**Pull request by tonybaloney**
Implements #1035 

- [x] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [x] Tests are included and/or updated for code changes.
- [x] Proper license headers are included in each file.

### CODE REVIEW

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (asw101) [inline src/startupscriptgenerator/src/python/scriptgenerator.go:347]: Will this handle paths with spaces?

**Comment by tonybaloney** [comment]
@karthiknadig please can you check the debug py command in https://github.com/microsoft/Oryx/pull/1036/files#diff-a47503bf32b2396a61d0f2ab7307019d0a6d4c9bd92b29dcf203f19cf275e73fR350 looks ok

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (karthiknadig) [inline src/startupscriptgenerator/src/python/scriptgenerator.go:350]: This looks right.

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (tonybaloney) [reply src/startupscriptgenerator/src/python/scriptgenerator.go:347]: I didn't see any shell escaping in any of the other examples. probably out of scope for this PR, but worth asking the maintainers.

**Comment by tonybaloney** [comment]
@microsoft/oryx-contrib can someone review this PR please

File: images/runtime/python/template.Dockerfile
Comment (arroyc) [inline images/runtime/python/template.Dockerfile:41]: we dont need ptvsd anymore i guess if we are adding debugpy

File: src/startupscriptgenerator/src/python/main.go
Comment (arroyc) [inline src/startupscriptgenerator/src/python/main.go:37]: remove ptvsd

File: src/startupscriptgenerator/src/python/main.go
Comment (arroyc) [inline src/startupscriptgenerator/src/python/main.go:125]: remove

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (arroyc) [inline src/startupscriptgenerator/src/python/scriptgenerator.go:43]: remove ptvsd?

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (arroyc) [inline src/startupscriptgenerator/src/python/scriptgenerator.go:122]: remove ptvsd as its deprecated?

**Review by arroyc — CHANGES_REQUESTED** [review]
I made some comments .. hopefully you can address those .. sorry about late response ..  currently we are short on people .. please expect some delay ... if you have urgent request please email

**Comment by tonybaloney** [comment]
@arroyc updates completed

**Review by arroyc — COMMENTED** [review]
this looks good now .. what i will do i will create a PR from this branch .. because external user's  pr do not trigger validation builds .. and without a successful build we cannot merge it to master

File: src/startupscriptgenerator/src/python/scriptgenerator.go
Comment (qianz2) [inline src/startupscriptgenerator/src/python/scriptgenerator.go:118]: 👍

**Comment by qianz2** [comment]
> this looks good now .. what i will do i will create a PR from this branch .. because external user's pr do not trigger validation builds .. and without a successful build we cannot merge it to master

Hey Arjun, I'm using the last commit id from this PR to manually trigger a validation build, it somehow worked. So let's wait until it passed the build, then we should be able to merge it, you don't need to create another branch for it.

**Review by qianz2 — APPROVED** [review]

**Comment by arroyc** [comment]
> > this looks good now .. what i will do i will create a PR from this branch .. because external user's pr do not trigger validation builds .. and without a successful build we cannot merge it to master
> 
> Hey Arjun, I'm using the last commit id from this PR to manually trigger a validation build, it somehow worked. So let's wait until it passed the build, then we should be able to merge it, you don't need to create another branch for it.

It didn't work, so we have to create a separate branch to merge it in.

**Comment by arroyc** [comment]
closing because of #1050

**Pull request closed without merging**
